crates/ruff_linter/src/rules/airflow/rules/removal_in_3.rs

```diff
@@ -71,6 +71,104 @@ impl Violation for Airflow3Removal {
     }
 }
 
+const REMOVED_CONTEXT_KEYS: [&str; 12] = [
+    "conf",
+    "execution_date",
+    "next_ds",
+    "next_ds_nodash",
+    "next_execution_date",
+    "prev_ds",
+    "prev_ds_nodash",
+    "prev_execution_date",
+    "prev_execution_date_success",
+    "tomorrow_ds",
+    "yesterday_ds",
+    "yesterday_ds_nodash",
+];
+
+fn extract_name_from_slice(slice: &Expr) -> Option<String> {
+    match slice {
+        Expr::StringLiteral(ExprStringLiteral { value, .. }) => Some(value.to_string()),
+        _ => None,
+    }
+}
+
+/// Checks for the use of deprecated Airflow context variables.
+///
+/// The function handles context keys accessed:
+/// - Directly using `context["key"]` or context.get("key").
+/// - Using `.get()` with variables derived from `get_current_context()`.
+/// - In custom operators, task decorators, and within templates or macros.
```

Comment:
Do we need to update the documentation for this function as it's only checking subscript expressions?

Reply:
I felt it is easier for someone to read the code logic in this function this way.

Reply:
I think we should keep the documentation limited to what the function is doing, we can add this as comments wherever it's relevant.

Reply:
Changed it now